network-libp2p/src/connection_pool/behaviour.rs

```diff
@@ -821,6 +793,89 @@ impl Behaviour {
             }
         }
     }
+
+    fn on_listen_failure(&mut self, send_back_addr: &Multiaddr) {
+        // Get IP from multiaddress if it exists.
+        let ip_info = self.get_ip_info_from_multiaddr(send_back_addr);
+
+        // Decrement IP counters if needed
+        if let Some(ip_info) = ip_info {
+            self.decrement_ip_counters(&ip_info);
+        }
+        self.limits.peer_count = self.limits.peer_count.saturating_sub(1);
+    }
+
+    fn reached_peer_limit(&mut self) -> Result<(), ConnectionDenied> {
```

Comment:
I'd rather have it named sth like this. For as it is currently named I would expect `OK(())` to denote that the limit was reached. On first glance I though this was handling the case where the limit was reached already.
```suggestion
    fn increase_and_check_counter(&mut self) -> Result<(), ConnectionDenied> {
```